src/key.rs

```diff
@@ -162,9 +163,8 @@ impl SecretKey {
     /// ```
     #[inline]
     pub fn from_slice(data: &[u8])-> Result<SecretKey, Error> {
-        match data.len() {
```

Comment:
In ac8e966ab5c4b0e6eb3c918e283aff0d3883427b:

Maybe worth thinking about turning these `from_slice` methods into `TryFrom` impls entirely ... doesn't need to be in this PR.

Reply:
FWIW I rekon we should keep this PR as lean as possible to help expedite review/merge.